AllMusicScripts/allMusicSraping.js

```diff
@@ -0,0 +1,248 @@
+const fs = require('fs');
+const request = require('request');
+const cheerio = require('cheerio');
+const sleep = require('system-sleep');
+const strSplit = rdateTimeequire('strsplit');
+var jsesc = require('jsesc');
+var newObj = [];
+
+function replaceAll(str, find, replace) {
+    return str.replace(new RegExp(find, 'g'), replace);
+}
+
+function replaceURLAndUnderscore(str) {
+    str = str.replace("http://www.allmusic.com/composition/", "")
+    return replaceAll(str, "_", " ")
+}
+//Extracted information  is in the form
+//Eg: December 16, 1770 in Bonn, Germany so we extract tokens from text.
+function getPlace(birthPlaceInString, callback) {
+    var birthPlace;
+    var values = []
+    var dataOfBirth = birthPlaceInString.split(",").map(function(val) {
+        values.push(val);
+    });
+    if (values[1]) {
+        var year = [];
+        strSplit(values[1].trim(), /\s+/).map(function(element) {
+            year.push(element);
+        });
+        if (year[2]) {
+            if (year[3]) {
+                birthPlace = year[2] + year[3];
+            } else {
+                birthPlace = year[2];
+            }
+        }
+    }
+    return callback(birthPlace);
+}
+
+function getDate(dateInString, callback) {
+    var date = [];
+    var values = [];
+    var dataOfBirth = dateInString.split(",").map(function(val) {
+        values.push(val);
+    });
+    month = []
+    if (values[0]) {
+        var dateOfBirth;
+        strSplit(values[0].trim(), /\s+/).map(function(ele) {
+            month.push(ele);
+        });
+        getMonth(month[0], function(monthNumber) {
+            var dayNum;
+            monName = monthNumber;
+            if (monName) {
+                dateOfBirth = month[1] + "-" + monName;
+                date.push(month[1]);
+                date.push(monName);
+            } else {}
+
+            if (values[1]) {
+                var year = [];
+                strSplit(values[1].trim(), /\s+/).map(function(element) {
+                    year.push(element);
+                });
+                date.push(year[0]);
+            }
+        });
+    }
+    return date;
+}
+
+function getMonth(monthName, callback) {
```

Comment:
this getMonth() function seems overly complicated?
Any reason, why you didn't just [use an object](http://stackoverflow.com/a/13566667/6621105)?